From a256bd0d0c29ec2fd7388c6ff9d110623cb075f9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 03 Sep 2021 18:59:50 +0300
Subject: [PATCH] [SW-0] ppboot: consolidate preboot_delay()

preboot_delay() function has real implementation only in Goya.
All the rest devices do nothing in this function.
Introduce a weak variant for all to stop copy-pasting empty code.

Change-Id: Ie7d65e5402ce9819c1de27edcb19c04ea5f15520
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 3a3901b..a5d3c9f 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -235,6 +235,7 @@
 
 __weak void ppboot_init_stack_mon_dev_specific(void) {}
 __weak void ppboot_check_stack_mon_dev_specific(void) {}
+__weak void preboot_delay(void) {}
 
 void ppboot_init_stack_monitor(void)
 {
diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index eb7a09c..c3958df 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -392,8 +392,6 @@
 	regmap_writel(rst_ctrl, mmCPU_CA53_CFG_ARM_RST_CONTROL);
 }
 
-void preboot_delay(void) {}
-
 #define mmPCIE_WRAP_VUART_TX_DATA	0xC01F60
 #define mmPCIE_WRAP_VUART_TX_STATUS	0xC01F64
 void preboot_putc_virt_uart(const char c)
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 68bffb6..886d5e0 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -416,8 +416,6 @@
 	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
 }
 
-void preboot_delay(void) {}
-
 unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_19;
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 5075be4..aded8a6 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -128,8 +128,6 @@
 	return mmPSOC_EFUSE_READ;
 }
 
-void preboot_delay(void) {}
-
 unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_19;
